### DISCUSSION TITLE
Making canvas fill the size of the parent

### DISCUSSION THREAD

**Discussion by runrc** [cogentcore/core#1126] [Q&A] [answered]
Running the code below, I would expect the canvas to fill the entire region of the parent and thus the line would be drawn from the top left (0,0) to the bottom right (1,1) of the parent (main window) However, this is not the case, the line is drawn and only covers about 40% of the region. 

How can I make it fill the entire region of the parent?

```
b := core.NewBody("Test")
core.NewCanvas(b).SetDraw(func(pc *paint.Context) {
  pc.MoveTo(0, 0)
  pc.LineTo(1, 1)
  pc.StrokeStyle.Color = colors.Scheme.Error.Base
  pc.Stroke()
})
b.RunMainWindow()
```

**Comment by kkoreilly** [comment, marked as answer]
This should work:

```go
b := core.NewBody("Test")
c := core.NewCanvas(b)
c.Styler(func(s *styles.Style) {
  s.Grow.Set(1, 1) // << this makes it take up the entire space
})
c.SetDraw(func(pc *paint.Context) {
  pc.MoveTo(0, 0)
  pc.LineTo(1, 1)
  pc.StrokeStyle.Color = colors.Scheme.Error.Base
  pc.Stroke()
})
b.RunMainWindow()
```

Depending on your use case, you might also be interested in our [data plots](https://www.cogentcore.org/core/widgets/other/plots).

**Comment by runrc** [comment]
Yes the above works. Thank you.
You may want to update your documentation for future reference.

**Reply by kkoreilly** [reply]
I updated the documentation accordingly in ddb85a0. If there are any features you would like added to our data plots, feel free to let me know. There are a lot of customization options and we are happy to add more.
